Xamarin.Essentials/FileSystem/FileSystem.ios.cs

```diff
@@ -176,25 +179,62 @@ internal override Task<Stream> PlatformOpenReadAsync()
         }
     }
 
+    // TODO: Think if we should rename it as UIImage is just an input
     class UIImageFileResult : FileResult
     {
-        readonly UIImage uiImage;
-        NSData data;
-
-        internal UIImageFileResult(UIImage image)
+        internal UIImageFileResult(UIImage image, NSDictionary metadata)
             : base()
         {
-            uiImage = image;
-
-            FullPath = Guid.NewGuid().ToString() + FileSystem.Extensions.Png;
-            FileName = FullPath;
+            FileName = Guid.NewGuid() + FileSystem.Extensions.Jpg;
+            var tempDir = NSFileManager.DefaultManager.GetTemporaryDirectory().Path;
+            FullPath = Path.Combine(tempDir, FileName);
+            SaveImageWithMetadata(image, metadata);
         }
 
         internal override Task<Stream> PlatformOpenReadAsync()
         {
-            data ??= uiImage.AsPNG();
+            Stream fileStream = File.OpenRead(FullPath);
+
+            return Task.FromResult(fileStream);
+        }
+
+        void SaveImageWithMetadata(UIImage image, NSDictionary metadata)
+        {
+            try
+            {
+                var imageData = image.AsJPEG();
 
-            return Task.FromResult(data.AsStream());
+                if (imageData == null)
+                {
+                    // TODO: Throw custom exception
+                }
+
+                // Copy over meta data
+                using var ciImage = CIImage.FromData(imageData);
+                var originalColorSpace = ciImage.ColorSpace;
+                using var newImageSource = ciImage.CreateBySettingProperties(metadata);
+                using var ciContext = new CIContext();
+                var result = ciContext.WriteJpegRepresentation(newImageSource, NSUrl.FromFilename(FullPath), originalColorSpace ?? CGColorSpace.CreateSrgb(), new NSDictionary(), out var error);
+                if (result)
+                {
+                    if (error != null)
+                    {
+                        // TODO: Throw custom exception and include the error
```

Comment:
You can throw an NSErrorException with the error.

Reply:
The logic may cause this to never be reached:

> Return Value
> If true, file export succeeded. If false, examine the errorPtr parameter for possible failure reasons.

https://developer.apple.com/documentation/coreimage/cicontext/1642218-writejpegrepresentation

Reply:
> The logic may cause this to never be reached:
> 
> > Return Value
> > If true, file export succeeded. If false, examine the errorPtr parameter for possible failure reasons.
> 
> https://developer.apple.com/documentation/coreimage/cicontext/1642218-writejpegrepresentation

True. My mistake, that second prospect exception should be in `else` clause